package/kernel/cifsd/Makefile

```diff
@@ -0,0 +1,57 @@
+include $(TOPDIR)/rules.mk
+include $(INCLUDE_DIR)/kernel.mk
+
+PKG_NAME:=cifsd
+PKG_RELEASE:=1
+
+PKG_SOURCE_PROTO:=git
+PKG_SOURCE_URL:=https://github.com/namjaejeon/cifsd.git
+PKG_SOURCE_DATE:=2019-02-15
+PKG_SOURCE_VERSION:=daf14f74fd105a338dc438beab042054e9720c43
+PKG_MIRROR_HASH:=521585ebfda0ecc02372b1a38ebf762fbbcead6d0b754a47599a5bf6bfdb3fb6
+
+PKG_MAINTAINER:=Andy Walsh <andy.walsh44+github@gmail.com>
+PKG_LICENSE:=GPL-2.0
+PKG_LICENSE_FILES:=COPYING
+
+include $(INCLUDE_DIR)/package.mk
+
+define KernelPackage/fs-cifsd
+	SUBMENU:=Filesystems
+	TITLE:=CIFS/SMB kernel server support
+	FILES:=$(PKG_BUILD_DIR)/cifsd.$(LINUX_KMOD_SUFFIX)
+	DEPENDS+= \
+		+kmod-nls-base \
+		+kmod-nls-utf8 \
+		+kmod-crypto-md4 \
+		+kmod-crypto-md5 \
+		+kmod-crypto-hmac \
+		+kmod-crypto-ecb \
+		+kmod-crypto-des \
+		+kmod-crypto-sha256 \
+		+kmod-crypto-cmac \
+		+kmod-crypto-sha512 \
+		+kmod-crypto-aead \
+		+kmod-crypto-ccm
+	KCONFIG:= \
+		CONFIG_KEYS=y \
+		CONFIG_CRYPTO_ARC4=y
```

Comment:
CONFIG_CRYPTO_ARC4 is already compiled in in OpenWrt.

Reply:
Just to ensure it stays this way if something changes or was locally changed. It was not clear to me how those cases are handled, since i assumed if you compile a custom kernel this setting always ensures you get a compatible one if you select this package?

Reply:
https://github.com/openwrt/openwrt/blob/master/target/linux/generic/config-4.14#L880
https://github.com/openwrt/openwrt/blob/master/target/linux/generic/config-4.19#L920
Unless you deliberately remove these lines it'll always be included

Reply:
Ok i'm still confused, maybe you explain how things should work.

If you use `KCONFIG:=` and the package is selected than you always end up with a compatible kernel. So for this reason i don't understand why we would remove the options from the package? 

I understand that we use the `generic/config` template to ensure that those packages can be installed and work without needing a new kernel, but to me that's just a extra specific option we choose.

Is there any harm by including the needed options via `KCONFIG:=`? I find this more clear since it documents what options a module actually needs.

Reply:
I think the issue is that opkg doesn't know of kernel config depends which means that you can technically install a package which wont run on in the end unless whaterver you need is provided in a module.